lib/index.js

```diff
@@ -121,12 +121,19 @@ function Socket (type, opts) {
   }
 
   /* start listening for inbound messages */
-  if(this.af_domain === nn.AF_SP) {
-    if (this.receiver) this._startPollReceive();
+  if (this.af_domain === nn.AF_SP && this.receiver) {
+    this._startPollReceive();
   }
+
+  if (!opts.hasOwnProperty('objectMode')) {
+    opts.objectMode = true;
+  }
```

Comment:
Also, is that number configurable?  It seems arbitrary that `objectMode` translates to 16 msgs.  Is that 16 1B messages?  What about 1KB messages?